### ISSUE TITLE
KL-divergence loss

### ISSUE THREAD

**Issue by runxiao** [cdoersch/vae_tutorial#4]
hi, I find there maybe a issue in model prototxt about the KL-divergence loss bewteen Q(z|X) and P(z).
In the paper,  the KL-divergence of Enquation 7:

The first term is  trace of diagonal matrix and should be sum of all diagonal elements, ex: x1+x2+x3.
But in the model file implementation is sum of square in diagonal.

layer{
  name: "var"
  type: "Eltwise"
  bottom: "sd"
  bottom: "sd"
  top: "var"
  eltwise_param{
    operation: PROD
  }
  include {
    phase: TRAIN
  }
}

layer{
  name: "kldiv_plus_half"
  type: "Eltwise"
  bottom: "meansq"
  bottom: "var"
  bottom: "logsd"
  top: "kldiv_plus_half"
  eltwise_param{
    operation: SUM
    coeff: 0.5
    coeff: 0.5
    coeff: -1.0
  }
  include {
    phase: TRAIN
  }
}

That makes me some confuse.

**Comment by cdoersch** [comment]
The formula for the KL divergence involves the trace of the _covariance matrix_.  The neural network outputs the log standard deviation for each dimension; you need to square these to get the diagonal of the covariance matrix.

Let me know if that makes sense and I'll close this bug.
